toolchains/prebuilt_toolchains.py

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+from textwrap import indent
+import urllib.request
+import hashlib
+
+
+CMAKE_SHA256_URL_TEMPLATE = "https://cmake.org/files/v{minor}/cmake-{full}-SHA-256.txt"
+CMAKE_URL_TEMPLATE = "https://github.com/Kitware/CMake/releases/download/v{full}/{file}"
+
+CMAKE_VERSIONS = [
+    "3.19.4",
+    "3.18.6",
+    "3.17.5",
+    "3.16.9",
+    "3.15.7",
+    "3.14.7",
+]
+
+CMAKE_TARGETS = {
+    "Darwin-x86_64": [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:macos",
+    ],
+    "Linux-aarch64": [
+        "@platforms//cpu:aarch64",
+        "@platforms//os:linux",
+    ],
+    "Linux-x86_64": [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:linux",
+    ],
+    "macos-universal": [
+        "@platforms//os:macos",
+    ],
+    "win32-x86": [
+        "@platforms//cpu:x86_32",
+        "@platforms//os:windows",
+    ],
+    "win64-x64": [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:windows",
+    ],
+}
+
+NINJA_URL_TEMPLATE = "https://github.com/ninja-build/ninja/releases/download/v{full}/ninja-{target}.zip"
+
+NINJA_TARGETS = {
+    "linux": [
+        "@platforms//os:linux",
+    ],
+    "mac": [
+        "@platforms//os:macos",
+    ],
+    "win": [
+        "@platforms//os:windows",
+    ],
```

Comment:
I also assume that this build is x86_64 (or maybe x86_32?) only and wouldn't work on arm64 windows.